template/config/clean.sls

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
+{#- Get the `topdir` from `tpldir` #}
+{%- set topdir = tpldir.split('/')[0] %}
+{%- set sls_service_clean = topdir ~ '.service.clean' %}
+{%- from topdir ~ "/map.jinja" import template with context %}
+{%- from topdir ~ "/macros.jinja" import files_switch with context %}
+
+include:
+  - {{ sls_service_clean }}
+
+template-config-absent:
```

Comment:
For predictable API a useful naming convention is to align state names with formula structure - so `template-config-absent` becomes `template-config-clean-absent` or `template config clean absent` in natural language.  

The convention scales well and and is nice when looping:

```
{%- for x in list %}
template config clean {{ x }} absent:
{%- endfor %}
```

Reply:
OK, I'll look at this as well.

Reply:
Done.

For a fully dependable, consistent structure, I saw 5 parts to the state ID naming:

* 3 parts from the filename:
    * `template/package/install.sls`
* 2 parts from the state:
    * `pkg.installed`
* In the structure:
    * `<formula>-<subdir1>-<subdir2>-<state1>-<state2>`
* Putting together to get:
    * `template-package-install-pkg-installed`

So across the whole formula:

| File                         | State           | State ID                                 |
|------------------------------|-----------------|------------------------------------------|
| template/package/install.sls | pkg.installed   | template-package-install-pkg-installed   |
| template/package/clean.sls   | pkg.removed     | template-package-clean-pkg-removed       |
| template/config/file.sls     | file.managed    | template-config-file-file-managed        |
| template/config/clean.sls    | file.absent     | template-config-clean-file-absent        |
| template/service/running.sls | service.running | template-service-running-service-running |
| template/service/clean.sls   | service.dead    | template-service-clean-service-dead      |